modules/phase_field/src/actions/PolycrystalColoringICAction.C

```diff
@@ -19,20 +19,29 @@ InputParameters
 PolycrystalColoringICAction::validParams()
 {
   InputParameters params = Action::validParams();
-  params.addClassDescription("Random Voronoi tessellation polycrystal action");
+  params.addClassDescription(
+      "Action to create ICs for polycrystal variables defined using a Voronoi tesselation");
   params.addRequiredParam<unsigned int>("op_num", "number of order parameters to create");
   params.addRequiredParam<std::string>("var_name_base", "specifies the base name of the variables");
   params.addRequiredParam<UserObjectName>("polycrystal_ic_uo", "Optional: TODO");
   params.addParam<std::vector<SubdomainName>>("block",
                                               "Block restriction for the initial condition");
+  params.addParam<bool>("nonlinear_preconditioning",
+                        false,
+                        "Whether to use nonlinear preconditioning or the standard model");
+  params.addParam<Real>("bound_value",
+                        5.0,
+                        "ONLY FOR NONLINEAR PRECONDITIONING: Bound value used to keep variable "
+                        "between +/-bound. Must be positive.");
 
   return params;
 }
 
 PolycrystalColoringICAction::PolycrystalColoringICAction(const InputParameters & params)
   : Action(params),
     _op_num(getParam<unsigned int>("op_num")),
-    _var_name_base(getParam<std::string>("var_name_base"))
+    _var_name_base(getParam<std::string>("var_name_base")),
+    _nonlin_precond(getParam<bool>("nonlinear_preconditioning"))
```

Comment:
I'd love to see all instances of "nonlin precond" expanded to "nonlinear preconditioning"

Reply:
@dschwen do you mean just variable names like here or everywhere, including the file names?

Reply:
@dschwen @laagesen We talked in person about changing the name of this because "nonlinear preconditioning" might make users think this is related to preconditioning in PETSc. @laagesen suggested "change-of-variable" but I think that might be too generic. Here are some other options I have thought of:

- Linear Interface
- Linear Substitution
- Linear Variable Substitution

Which do you prefer?

Reply:
The substitution is not linear, but it _linearizes_ the order parameters. `LinearizedInterface` would be my sugegstion.

Reply:
I like LinearizedInterface

Reply:
I also like `LinearizedInterface`.